Doc/library/typing.rst

```diff
@@ -874,10 +874,15 @@ using ``[]``.
               case _:
                   never_call_me(arg)  # OK, arg is of type Never
 
+   ``Never`` can also be used to indicate that a function never returns,
+   but ``NoReturn`` is more explicit.
```

Comment:
This reads like a style suggestion; I'm not sure CPython should be suggesting style here (and if anything I'd prefer the opposite style)

Reply:
Quoting @JelleZijlstra:

> `Never` and `NoReturn` are exactly equivalent from a type checker's perspective and we should make the docs say so clearly. I suppose if we have to give a difference, we can say that **`NoReturn` can be preferred for return types and `Never` for other places.**

Either we say that, or we give an example and deprecate `NoReturn`. Otherwise, we have two similar ways to mark functions that do not return, making `NoReturn` redundant.

Reply:
I think we should just deprecate `NoReturn` and remove it in 3.18.

Reply:
@AlexWaygood, then what *do* we do instead?

Reply:
I have heard your question and I will respond when I have a moment to sit down and think about it